pkg/ldap/datasource.go

```diff
@@ -826,8 +826,7 @@ func StringAttrValuesToRequestedType(
 
 		return value, nil
 	case getAttrType(api_adapter_v1.AttributeType_ATTRIBUTE_TYPE_INT64):
-		// All numbers are unmarshalled into float64. Further framework converts into int64.
-		value, err := strconv.ParseFloat(attr.Values[0], 64)
+		value, err := strconv.ParseInt(attr.Values[0], 10, 64)
```

Comment:
Can you remove the TODO on line 816? `// TODO: optimize case of DOUBLE & INT64 type`